src/core/namespace.c

```diff
@@ -887,9 +944,15 @@ static int make_read_only(const MountEntry *m, char **blacklist, FILE *proc_self
         assert(m);
         assert(proc_self_mountinfo);
 
-        if (mount_entry_read_only(m))
+        if (mount_entry_read_only(m)) {
+                if (IN_SET(m->mode, EMPTY_DIR, TMPFS))
+                        /* also make superblock readonly */
+                        if (mount(NULL, mount_entry_path(m), NULL, MS_REMOUNT | MS_RDONLY | m->flags, mount_entry_options(m)) < 0)
+                                if (errno != ENOENT || !m->ignore)
+                                        return -errno;
+
```

Comment:
(the call to bind_remount_recursive_with_mountinfo() below:) ah, hmm, doesn't this mean the `ro` tmpfs will also remount all the bind mounts which are underneath it to become read-only bind mounts?  I don't think we want that.

Reply:
Ah, I see.  I missed the `blacklist` parameter that makes this work correctly.

Reply:
I still think this will work fine without the second, MS_BIND remount of the tmpfs.  Having both calls shouldn't break anything though.

Reply:
OK, now I agree with you. Calling `bind_remount_recursive_with_mountinfo()` for TMPFS and EMPTY_DIR is dropped.